packages/prettier-plugin-java/src/printers/printer-utils.js

```diff
@@ -531,6 +531,63 @@ function isStatementEmptyStatement(statement) {
   );
 }
 
+function sortImports(imports) {
+  const staticImports = [];
+  const nonStaticImports = [];
+
+  if (imports !== undefined) {
+    for (let i = 0; i < imports.length; i++) {
+      if (imports[i].children.Static !== undefined) {
+        staticImports.push(imports[i]);
+      } else if (imports[i].children.emptyStatement === undefined) {
+        nonStaticImports.push(imports[i]);
+      }
+    }
+
+    // TODO: Could be optimized as we could expect that the array is already almost sorted
+    staticImports.sort((first, second) =>
+      compareFqn(
+        first.children.packageOrTypeName[0],
+        second.children.packageOrTypeName[0]
+      )
+    );
+    nonStaticImports.sort((first, second) =>
+      compareFqn(
+        first.children.packageOrTypeName[0],
+        second.children.packageOrTypeName[0]
+      )
```

Comment:
Maybe you could factor this by declaring the comparator in a variable and then use it?
```
const comparator = (first, second) =>
      compareFqn(
        first.children.packageOrTypeName[0],
        second.children.packageOrTypeName[0]
      );
staticImports.sort(comparator);
nonStaticImports.sort(comparator);
```